metis-authentication/metis-authentication-service/src/main/java/eu/europeana/metis/authentication/dao/ZohoAccessClientDao.java

```diff
@@ -36,40 +44,79 @@ public ZohoAccessClientDao(String zohoBaseUrl, String zohoAuthenticationToken) {
 
   public JsonNode getUserByEmail(String email) throws IOException {
     String contactsSearchUrl = String
-        .format("%s/%s/%s/%s", zohoBaseUrl, "json", CONTACTS_MODULE_STRING, SEARCH_RECORDS_STRING);
+        .format("%s/%s/%s/%s", zohoBaseUrl, JSON_STRING, CONTACTS_MODULE_STRING,
+            SEARCH_RECORDS_STRING);
     UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(contactsSearchUrl)
         .queryParam(AUTHENTICATION_TOKEN_STRING, zohoAuthenticationToken)
-        .queryParam(SCOPE_STRING, "crmapi")
-        .queryParam(CRITERIA_STRING, String.format("(Email:%s)", email));
+        .queryParam(SCOPE_STRING, CRMAPI_STRING)
+        .queryParam(CRITERIA_STRING, String.format("(%s:%s)", EMAIL_FIELD, email));
 
     RestTemplate restTemplate = new RestTemplate();
     String contactResponse = restTemplate
         .getForObject(builder.build().encode().toUri(), String.class);
     LOGGER.info(contactResponse);
     ObjectMapper mapper = new ObjectMapper();
     JsonNode jsonContactResponse = mapper.readTree(contactResponse);
-    if (jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING) == null)
+    if (jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING) == null) {
       return null;
-    return jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING).get(CONTACTS_MODULE_STRING).get("row").get("FL");
+    }
+    return jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING).get(CONTACTS_MODULE_STRING)
+        .get(ROW_STRING).get(FIELDS_LABEL);
   }
 
   public JsonNode getOrganizationByOrganizationName(String organizationName) throws IOException {
```

Comment:
I suggest returning only the organization ID as this is the only thing that is needed and do the check for the presence of an organization role (see MetisUser.setAndCheckOrganizationIdFromJsonNode()) somewhere in this class instead.